### PR TITLE
Refactor Default Config [API-1376]

### PR DESCRIPTION
**Pull request by utku-caglayan**
Refactors the default config file that the user interacts with.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -70,13 +70,35 @@ func DefaultConfig() *Config {
...
-	var out []byte
-	if out, err = yaml.Marshal(config); err != nil {
-		return err
-	}
-	return file.CreateMissingDirsAndFileWithRWPerms(confPath, out)
+const defaultUserConfig = `hazelcast:
+  clientname: ""
+  logger:
+    level: error
+  cluster:
```
Comment (yuce) [inline config/config.go:77]: Could you also add the `network` config? Probably `addresses` is adequate.

File: config/config.go [hunk 2]
```diff
@@ -70,13 +70,35 @@ func DefaultConfig() *Config {
...
+    cloud:
+      token: ""
+      enabled: false
+    discovery:
+      usepublicip: false
+    unisocket: true
+ssl:
+  enabled: false
+  servername: ""
+  insecureskipverify: false
```
Comment (yuce) [inline config/config.go:92]: IMO we can remove `insecurekipverify`. The user can check it in the docs.

**Review by yuce — COMMENTED** [review]
Looks good! Just two ideas.

File: config/config.go (hunk 2 above)
Comment (utku-caglayan) [reply config/config.go]: Good idea, removed it on https://github.com/hazelcast/hazelcast-commandline-client/pull/103/commits/d6297f724be67ca4d064e4783dae5e668bfba38b

File: config/config.go (hunk 1 above)
Comment (utku-caglayan) [reply config/config.go:77]: Added it here with connection timeout field https://github.com/hazelcast/hazelcast-commandline-client/pull/103/commits/d6297f724be67ca4d064e4783dae5e668bfba38b

**Review by yuce — APPROVED** [review]
Looks good!

File: config/config.go [hunk 3]
```diff
@@ -70,13 +70,37 @@ func DefaultConfig() *Config {
...
+        password: ""
+    name: dev
+    cloud:
+      token: ""
+      enabled: false
+    discovery:
+      usepublicip: false
+    unisocket: true
+  network:
+      addresses: []
```
Comment (yuce) [inline config/config.go:90]: Could you change this one to:
```
addreses:
    "localhost:5701"
```

File: config/config.go [hunk 4]
```diff
@@ -70,13 +70,37 @@ func DefaultConfig() *Config {
...
+      connectiontimeout: 0s
```
Comment (yuce) [inline config/config.go:91]: I think we need a comment on this config item that explains `0s == infinity`

**Review by yuce — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 3 above)
Comment (utku-caglayan) [reply config/config.go]: Makes sense, thanks! https://github.com/hazelcast/hazelcast-commandline-client/pull/103/commits/e36873bca1e2ed61955ab506ce8f43f9152267a7

File: config/config.go (hunk 4 above)
Comment (utku-caglayan) [reply config/config.go]: I added a comment but not sure if it is ok. I'd love to hear your suggestions if you have any https://github.com/hazelcast/hazelcast-commandline-client/pull/103/commits/e36873bca1e2ed61955ab506ce8f43f9152267a7

**Review by yuce — APPROVED** [review]
Looks good!

**Review by mehmettokgoz — COMMENTED** [review]
Looks good, thanks!

**Pull request merged**